bdns/dns.go

```diff
@@ -221,8 +229,20 @@ func New(
 	)
 	stats.MustRegister(queryTime, totalLookupTime, timeoutCounter, idMismatchCounter)
 
+	var client exchanger = dnsClient
+	if features.Enabled(features.DOH) {
+		client = &dohExchanger{
+			hc: http.Client{
+				Timeout: readTimeout,
+				Transport: &http.Transport{
+					TLSClientConfig: tlsConfig,
+				},
+			},
+		}
+	}
```

Comment:
Why not do this up at lines 194-198, so that we don't even have to call `new(dns.Client)` if features.DOH is enabled?